presto-main/src/main/java/io/prestosql/execution/SqlQueryManager.java

```diff
@@ -316,4 +326,21 @@ private void enforceCpuLimits()
             }
         }
     }
+
+    /**
+     * Enforce max raw data size at the query level
+     */
+    public void enforceQueryMaxDataSizeLimits()
+    {
+        for (QueryExecution query : queryTracker.getAllQueries()) {
+            if (query.getState().isDone()) {
+                continue;
+            }
+            DataSize queryMaxDataSize = SystemSessionProperties.getQueryMaxDataSize(query.getSession());
```

Comment:
You can go for a static import of `getQueryMaxDataSize`

Reply:
done